src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -32,193 +32,668 @@
+                fieldMapping.setObjectMapping(existingDefaultObjectMapping);
+                // Setting name to null will force field mapping to generate a unique name
+                fieldMapping.setRecordName(null, false);
+
+                queuedMetadataTypesForDeploy.add(fieldMapping.getMetadataRecord());
+
+            } else if (existingDefaultFieldMapping != null) {
+
+                if (customFieldMappingSetExists) {
+                    // Migrated Custom Field Mapping Set exists
+                    DataImportFieldMapping overwrittenMapping =
+                        new DataImportFieldMapping(
+                            existingDefaultFieldMapping,
+                            MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
+
+                    overwrittenMapping.setFieldMappingSet(MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
+                    overwrittenMapping.setObjectMapping(existingDefaultObjectMapping);
+                    overwrittenMapping.setTargetField(fieldMapping.targetFieldAPIName);
+
+                    queuedMetadataTypesForDeploy.add(overwrittenMapping.getMetadataRecord());
+                } else {
+                    // Check if we have a conflict and update the target field based on the help text mapping
+                    if (fieldMapping.hasBadDefaultValues(existingDefaultFieldMapping)) {
+                        fieldMapping.setFieldMappingSet(MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
+                        fieldMapping.setObjectMapping(existingDefaultObjectMapping);
+                        fieldMapping.setTargetField(fieldMapping.targetFieldAPIName);
+                    } else {
+                        fieldMapping.setFieldMappingSet(MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
+                        fieldMapping.setObjectMapping(existingDefaultObjectMapping);
+                    }
+
+                    queuedMetadataTypesForDeploy.add(fieldMapping.getMetadataRecord());
+                }
+            }
+
+            fieldMappingsByFieldMappingSetDeveloperName.remove(fieldMappingKey);
+        }
+
+        createAndEnqueueNonHelpTextMappings(fieldMappingsByFieldMappingSetDeveloperName);
+        createAndEnqueueMigratedCustomFieldMappingSet();
     }
 
-    /**
-     * @description Collect all Help Text field mappings and compare against default Field Mapping Metadata 
-     * Type records reserve all non-standard NPSP Help Text mappings. What to do with reserved list of Help 
-     * Text mappings TBD.
+    /*******************************************************************************************************
+     * @description Go through all fields in Data Import object and populate relevant lists from help text
+     * mappings
      */
-    public void buildCustomMetadataRecordsFromHelpText(BDI_MigrationMappingHelper helper) {
+    public void populateListsFromHelpTextMappings() {
```

Comment:
This method should be private since it is only used in the Utility class, with an @TestVisible tag so that it can be called from the tests.